python_modules/dagster/dagster/_core/definitions/module_loaders/utils.py

```diff
@@ -1,19 +1,49 @@
 import pkgutil
 from importlib import import_module
 from types import ModuleType
-from typing import Iterable, Iterator, Mapping, Tuple, Type, Union
+from typing import Any, Iterable, Iterator, Mapping, Tuple, Type, Union
 
 from dagster._core.definitions.asset_key import AssetCheckKey, AssetKey
 from dagster._core.definitions.asset_spec import AssetSpec
 from dagster._core.definitions.assets import AssetsDefinition
 from dagster._core.definitions.cacheable_assets import CacheableAssetsDefinition
+from dagster._core.definitions.executor_definition import ExecutorDefinition
+from dagster._core.definitions.job_definition import JobDefinition
+from dagster._core.definitions.logger_definition import LoggerDefinition
+from dagster._core.definitions.partitioned_schedule import (
+    UnresolvedPartitionedAssetScheduleDefinition,
+)
+from dagster._core.definitions.schedule_definition import ScheduleDefinition
+from dagster._core.definitions.sensor_definition import SensorDefinition
 from dagster._core.definitions.source_asset import SourceAsset
+from dagster._core.definitions.unresolved_asset_job_definition import UnresolvedAssetJobDefinition
+from dagster._core.executor.base import Executor
 
 LoadableAssetObject = Union[AssetsDefinition, AssetSpec, SourceAsset, CacheableAssetsDefinition]
-LoadableDagsterObject = LoadableAssetObject  # For now
+ScheduleDefinitionObject = Union[ScheduleDefinition, UnresolvedPartitionedAssetScheduleDefinition]
+JobDefinitionObject = Union[JobDefinition, UnresolvedAssetJobDefinition]
+ExecutorObject = Union[ExecutorDefinition, Executor]
+LoggerDefinitionKeyMapping = Mapping[str, LoggerDefinition]
+ResourceDefinitionMapping = Mapping[str, Any]
+LoadableDagsterObject = Union[
+    LoadableAssetObject,
+    SensorDefinition,
+    ScheduleDefinitionObject,
+    JobDefinitionObject,
+    ExecutorObject,
+]
 RuntimeKeyScopedAssetObjectTypes = (AssetsDefinition, AssetSpec, SourceAsset)
 RuntimeAssetObjectTypes = (AssetsDefinition, AssetSpec, SourceAsset, CacheableAssetsDefinition)
-RuntimeDagsterObjectTypes = RuntimeAssetObjectTypes  # For now
+RuntimeScheduleObjectTypes = (ScheduleDefinition, UnresolvedPartitionedAssetScheduleDefinition)
```

Comment:
This structure looks pretty annoying to maintain.

I've run into this annoyance myself, but it does look like `get_args(Union[x, y, z])` (imported from typing) would return a tuple of (x, y, z), which we could use for the runtime type checking, which would eliminate this sort of duplication

Reply:
ah nice. That seems acceptable to me.

Reply:
just to follow up on this; I think that the get_args approach doesn't actually work with type introspection in pylance; so I think this solution is no longer acceptable. I agree that it's annoying to maintain but I don't think it's _that_ annoying